### ISSUE TITLE
[Question] Aria-rowindex with advanced grid features

### ISSUE THREAD

**Issue by jivanova** [w3c/aria-practices#1275]
I've seen in the repo a question regarding filtering, where @jnurthen has explained that the `aria-rowindex` should be recalculated.

https://github.com/w3c/aria-practices/issues/465

1) Filtering & Sorting (just to clarify)

Applying filter, or sort should recalculate the `aria-rowindex` in the data grid. Meaning that the sequence would be always from index number 1, and there should not be confusing order in the table.

2) Paging

![image](https://user-images.githubusercontent.com/7159157/69323105-bd3b8b80-0c4e-11ea-8657-3580c7f6f79f.png)

Having the above image, what would we expect the sequence to be for the pages? Should we reset the `aria-rowindex` to start from one, or should show the actual data item index. We always have the `tr`s rendered only for the selected page.

3) Endless scrolling

Having dataitems, loaded while scrolling or navigating with the arrow keys. The data grid renders only the visible `tr`s, meaning that it's the same concept as in the Paging section. 

For the 3) point, I believe that it will be confusing to reset the row index as we support going to the next rows with the arrow keys, or with whatever keyboard navigation. As for the 2) point, I have my doubts, and I would really appreciate any feedback and opinion that you might share.

Thanks in advance!
Joana Ivanova

**Comment by jnurthen** [comment]
1. Filtering & Sorting (just to clarify)
  Yes - reset `aria-rowindex`
2. Paging
  Yes - reset `aria-rowindex` for each new page. Only one page of data is displayed at a time. Saying you are on row 30 when you are on page 3 would lead a user to expect they could move to a previous row in the grid by using up arrow even though the previous row is on page 2 of the data.
3. Endless scrolling
  No do not reset `aria-rowindex`, as to the user all of the rows are displayed. This is why my original reply had displayed in quotes. You should set aria-rowindex based on the user expectations of what is displayed not what is technically present in the DOM. If you are simulating a big grid only displaying a small subset of the rows this is exactly the situation `aria-rowindex` is made to help with.

**Comment by jivanova** [comment]
@jnurthen Thank you so much! Very useful and straight to the points reply...awesome!
